run_survey.py

```diff
@@ -1,90 +1,221 @@
+        )
 
-    with open(output_filepath, "w") as file:
-        json.dump(results, file, indent=4)
+    with open(
+        os.path.join(experiment_metadata_dir, "results.json"), "w"
+    ) as file:
+        json.dump(
+            results,
+            file,
+            indent=4,
+        )
+
+
+def count_off(seconds: int = 5):
+    for i in range(seconds):
+        print(seconds - i)
+        time.sleep(1)
 
 
-def main(
+async def main(
     model_name: str,
     survey_name: str,
+    experiment_name: str,
     n_samples_per_dependent_variable: typing.Optional[int] = None,
+    overwrite: bool = False,
+    infill: bool = False,
 ) -> None:
-    survey_directory = os.path.join("data", survey_name)
+    data_dir = os.path.join("data", survey_name)
+    variables_dir = os.path.join("variables", survey_name)
+    experiment_dir = os.path.join("experiments", experiment_name, survey_name)
 
-    with open(
-        os.path.join(survey_directory, "independent-variables.json"), "r"
-    ) as file:
-        independent_variable_names = json.load(file)
+    parsed_model_name = parse_model_name(model_name)
+    results_path = Path(experiment_dir) / parsed_model_name / "results.json"
 
-    with open(os.path.join(survey_directory, "dependent-variables.json"), "r") as file:
-        dependent_variable_names = json.load(file)
+    # Use pathlib to check if  a file exists
+    if results_path.is_file():
+        print(f"Experiment {experiment_name} already exists.")
 
-    survey = Survey(
-        name=survey_name,
-        data_filename=os.path.join(survey_directory, "data.csv"),
-        config_filename=os.path.join(survey_directory, "config.json"),
-        independent_variable_names=independent_variable_names,
-        dependent_variable_names=dependent_variable_names,
-    )
+        if overwrite and infill:
+            print("Choose whether to overwrite or infill. Exiting.")
+            return
+        elif infill:
+            print("\nINFILLING MISSING RESPONSE OBJECTS IN")
+            # count_off()
+        elif overwrite:
+            print("\nOVERWRITING IN")
+            # count_off()
+        else:
+            print(
+                "Use --force to overwrite or --infill to infill missing"
+                " response_objects. Exiting."
+            )
+            return
+    else:
+        if infill:
+            print(
+                "Cannot infill missing response objects if the experiment does"
+                " not exist. Exiting."
+            )
+            return
 
-    sampler = AutoSampler(model_name=model_name)
+    if logging:
+        logger = logging.getLogger(__name__)
+        logger.setLevel(logging.ERROR)
+        handler = logging.FileHandler(Path(experiment_dir) / "errors.log")
+        handler.setLevel(logging.ERROR)
+        formatter = logging.Formatter("%(asctime)s %(message)s")
+        handler.setFormatter(formatter)
+        logger.addHandler(handler)
+    else:
+        logger = None
```

Comment:
You can configure the global logger instead of passing a logger to every class. I'd say forcing every class to have a logger parameter is likely an antipattern too.